addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aAbstractHandler.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    @Override
+    public void dispose() {
+        houseUnitCode = "";
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        logger.debug("CM11A status changed to {}.", bridgeStatusInfo.getStatus());
+
+        if (bridgeStatusInfo.getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            logger.debug("CM11A is not online.", bridgeStatusInfo.getStatus());
+            return;
+        }
+
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            if (houseUnitCode.length() > 0) {
+                // The config must be present and was set during initialization
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            }
+        }
+    }
+
+    @Override
+    public abstract void handleCommand(ChannelUID channelUID, Command command);
+
+    /**
+     * Will be called by the X10Interface when it is ready for this X10 device to use the X10 bus.
+     * Child classes should override this method with the specific process necessary to update the
+     * hardware with the latest data.
+     *
+     * <p>
+     * Warning: This will be called in a different thread. It must be thread safe.
+     * </p>
+     *
+     * <p>
+     * Retries in the event of interface problems will be handled by the X10Interface. If a comms
+     * problem occurs and the method throws an exception, this device will be rescheduled again later.
+     * </p>
+     */
+    public abstract void updateHardware(X10Interface x10Interface) throws IOException, InvalidAddressException;
+
+    public State getCurrentState() {
+        return currentState;
+    }
+
+    public void setCurrentState(State currentState) {
+        this.currentState = currentState;
+    }
+
+    /**
+     * Subtract the specified number of X10 "dims" from the current state
+     *
+     * @param dims The number of dims to remove
+     * @return
+     */
+    public State addDimsToCurrentState(int dims) {
+        if (!(currentState instanceof PercentType)) {
+            if (currentState instanceof OnOffType && currentState == OnOffType.ON) {
+                currentState = PercentType.HUNDRED;
+            } else {
+                currentState = PercentType.ZERO;
+            }
+        }
+        // The current state is stored in a PercentType object and therefore needs to be converted to an incremental
+        // percent
+        int curPercent = ((PercentType) currentState).intValue();
+        // dims is a number between 0 and 22 which represents the full range of cm11a
+        int dimsPercent = (dims * 100) / X10_DIM_INCREMENTS;
+        int newPercent = curPercent - dimsPercent;
+        newPercent = Math.max(newPercent, 0);
+        newPercent = Math.min(newPercent, 100);
+        currentState = new PercentType(newPercent);
```

Comment:
Unneeded local before return

Reply:
Are you referring to currentState? This is a class variable. Maybe I am misunderstanding what the problem is, sorry.

Reply:
Nope, I was wrong :-)